src/Simplex/Messaging/Server/Main.hs

```diff
@@ -44,62 +56,99 @@ smpServerCLI cfgPath logPath =
     defaultServerPort = "5223"
     executableName = "smp-server"
     storeLogFilePath = combine logPath "smp-server-store.log"
-    initializeServer InitOptions {enableStoreLog, signAlgorithm, ip, fqdn} = do
-      deleteDirIfExists cfgPath
-      deleteDirIfExists logPath
-      createDirectoryIfMissing True cfgPath
-      createDirectoryIfMissing True logPath
-      let x509cfg = defaultX509Config {commonName = fromMaybe ip fqdn, signAlgorithm}
-      fp <- createServerX509 cfgPath x509cfg
-      writeFile iniFile iniFileContent
-      putStrLn $ "Server initialized, you can modify configuration in " <> iniFile <> ".\nRun `" <> executableName <> " start` to start server."
-      printServiceInfo serverVersion fp
-      warnCAPrivateKeyFile cfgPath x509cfg
+    initializeServer opts
+      | scripted opts = initialize opts
+      | otherwise = do
```

Comment:
For clarity there could be lines after each question:
"You selected: yes / no / random / etc."
 - just to be sure that you selected what was intended, e.g. after typing "y" instead of offered "Y".

Or/and print summary in the end. Not necessary though